bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/ObserveUtils.java

```diff
@@ -124,6 +125,37 @@ public static void startResourceObservation(Environment env, BString serviceName
         env.setStrandLocal(ObservabilityConstants.SERVICE_NAME, service);
     }
 
+    /**
+     * Add record checkpoint data to active Trace Span.
+     *
+     * @param env The Ballerina Environment
+     * @param pkg The package the instrumented code belongs to
+     * @param position The source code position the instrumented code defined in
+     */
+    public static void recordCheckpoint(Environment env, BString pkg, BString position) {
+        if (!tracingEnabled) {
+            return;
+        }
+
+        ObserverContext observerContext = (ObserverContext) env.getStrandLocal(KEY_OBSERVER_CONTEXT);
+        if (observerContext == null) {
+            return;
+        }
+        BSpan span = (BSpan) observerContext.getProperty(KEY_SPAN);
+        if (span == null) {
+            return;
+        }
+
+        // Adding Position and Module ID to the Jaeger Span
+        Map<String, String> eventAttributes = new HashMap<>(2);
+        eventAttributes.put(TAG_KEY_MODULE, pkg.getValue());
+        eventAttributes.put(TAG_KEY_INVOCATION_POSITION, position.getValue());
+
+        HashMap<String, Object> events = new HashMap<>(1);
```

Comment:
Declaration should be Map

Reply:
Usage of a HashMap has been removed with the migration to Opentelemetry ->https://github.com/ballerina-platform/ballerina-lang/commit/4b8659bb0bc82db1593ba1c008e25937dd9c075d